src/ReadingBuddy.java

```diff
@@ -391,16 +490,24 @@ public static ArrayList<String[]> addNewUserData(String uname, String pw, ArrayL
         return allUsers;
     }
 
-    public static void updateBookBuddysEntry(BookGroup groupInfo){
```

Comment:
this whole area of changes looks really weird/confusing but essentially:
- getSpecificCurrentBook is a completely new function
- the red chunk of code lines 396-401 are the previous/pre-update code for updateBookBuddysEntry (now starts line 505)